DAO.sol

```diff
@@ -511,9 +535,56 @@ contract DAO is DAOInterface, Token, TokenCreation {
             blocked[msg.sender] = _proposalID;
         }
 
+        votingRegister[msg.sender].push(_proposalID);
         Voted(_proposalID, _supportsProposal, msg.sender);
     }
 
+    function unVote(uint _proposalID){
```

Comment:
So then I think `unVote` should remove the proposal ID from `votingRegister[msg.sender]` Otherwise if someone votes on too many proposals it will result in them being unable to call `unVoteAll` due to block gas limit. Then they cannot call `unVote` to remove once blockage at a time since `votingRegister[msg.sender]` remains the same.